run.sh

```diff
@@ -3,13 +3,30 @@
 set -e
 
 ## If the mounted data volume is empty, populate it from the default data
+## The plugins.txt file lists plugins which should be installed.
 if ! [[ "$(ls -A /opt/sonarqube/data)" ]]; then
-	cp -a /opt/sonarqube/data-init /opt/sonarqube/data
+    cp -a /opt/sonarqube/data-init /opt/sonarqube/data
 fi
 
 ## If the mounted extensions volume is empty, populate it from the default data
 if ! [[ -d /opt/sonarqube/data/plugins ]]; then
 	cp -a /opt/sonarqube/extensions-init/plugins /opt/sonarqube/data/plugins
+	curl -s https://update.sonarsource.org/update-center.properties | grep downloadUrl > /tmp/pluginList.txt
+	printf "Downloading additional plugins\n"
+    for PLUGIN in $(echo $SONAR_PLUGIN_LIST)
+    do
+        DOWNLOAD_URL=$(cat /tmp/pluginList.txt | grep ${PLUGIN} | sort -V | tail -n 1 | awk -F"=" '{print $2}' | sed 's@\\:@:@g')
+
+        ## Check to see if plugin exists, attempt to download the plugin if it does exist.
+        if ! [[ -z "${DOWNLOAD_URL}" ]]; then
+            printf "    %-15s" ${PLUGIN}
+            curl -Ls -o /opt/sonarqube/data/plugins/${PLUGIN}.jar ${DOWNLOAD_URL} >> /dev/null 2>&1 && printf "%10s" "DONE" || printf "%10s" "FAILED"
+            printf "\n"
+        else
+            ## Plugin was not found in the plugin inventory
+            printf "    %-15s%10s\n" "${PLUGIN}" "NOT FOUND"
+        fi
+    done
 fi
```

Comment:
In OpenShift land, it would probably be ideal for this to be done via S2I, so that the resulting binary config is baked into the image, and the source config (e.g. plugin list) can be version controlled. Jenkins is a good example of that approach here: https://github.com/openshift/jenkins/tree/master/2/contrib/s2i

I'm fine with this approach thought for now, I just want you to be aware of this more generally. It's also possible I'm overlooking something here that make s2i problematic.